pkg/argocd/argocd.go

```diff
@@ -454,22 +465,24 @@ func SetKustomizeImage(app *v1alpha1.Application, newImage *image.ContainerImage
 
 	log.WithContext().AddField("application", app.GetName()).Tracef("Setting Kustomize parameter %s", ksImageParam)
 
-	if app.Spec.Source.Kustomize == nil {
-		app.Spec.Source.Kustomize = &v1alpha1.ApplicationSourceKustomize{}
+	// This case seems not possible, but we check it anyway
+	kustomizeSpec := app.Spec.GetSource().Kustomize
```

Comment:
In order to fix the error showed in the unit tests I should change some code back to the original and leave it like the suggestion below where the app.Spec.Source.Kustomize will be initialized:

```suggestion
	app.Spec.Source.Kustomize = app.Spec.GetSource().Kustomize
	if app.Spec.Source.Kustomize == nil {
		app.Spec.Source.Kustomize = &v1alpha1.ApplicationSourceKustomize{}
	}

	for i, kImg := range app.Spec.Source.Kustomize.Images {
		curr := image.NewFromIdentifier(string(kImg))
		override := image.NewFromIdentifier(ksImageParam)

		if curr.ImageName == override.ImageName {
			curr.ImageAlias = override.ImageAlias
			app.Spec.Source.Kustomize.Images[i] = v1alpha1.KustomizeImage(override.String())
		}

	}

	app.Spec.Source.Kustomize.MergeImage(v1alpha1.KustomizeImage(ksImageParam))

	return nil
```

Reply:
I added this two snippets to fix the errors showed by the unit tests and to keep the same code patterns on the Helm and Kustomize sections.

Reply:
Here we tried this and it worked to some degree. But in the end, if you want to set the target writeback repository is a bit messy. It would be better to use an annotation.

There is already a merge in the master branch of this repository that adds an annotation to set the writeback repository using multisource, but it seems to fail in the unit tests.

Digging a bit into the code, for this to work properly, it will need some dependencies upgrades (specially the argocd depedency) and a much deeper refactor and development to add this feature.

Reply:
I opened a new PR here (#636) with similar changes to this one but covering some issues that we found when we wanted to write back to Helm values files. I hope this helps.

Please, let me know if we should merge this to PR contents to have only one and work on that.